launch/launch/actions/execute_process.py

```diff
@@ -716,6 +735,31 @@ async def __execute_process(self, context: LaunchContext) -> None:
             self.__logger.error("process has died [pid {}, exit code {}, cmd '{}'].".format(
                 pid, returncode, ' '.join(cmd)
             ))
+            # respawn the process that abnormally died
+            if not context.is_shutdown and self.__respawn:
+                respawn_flag = False
+                if self.__respawn_delay > 0:
+                    try:
+                        # wait for a timeout(`self.__respawn_delay`) to respawn the process
+                        # and handle shutdown event with future(`self.__shutdown_future`)
+                        # to make sure `ros2 launch` exit in time
+                        await asyncio.wait_for(
+                            self.__shutdown_future,
+                            timeout=self.__respawn_delay,
+                            loop=context.asyncio_loop
+                        )
+                    except asyncio.exceptions.TimeoutError:
+                        respawn_flag = True
+                        self.__shutdown_future = create_future(context.asyncio_loop)
```

Comment:
@iuhilnehc-ynos I don't follow why this future has to be recreated.

Reply:
@hidmic 
I think it's because I misuse the **asyncio.wait_for** that will cancel the future inside if timeout, so it needs to create_futurn again. Anyway, I'll update it by using **asyncio.wait** with **[asyncio.sleep, self.__shutdown_future]**.